sql/api/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryListener.scala

```diff
@@ -146,14 +156,17 @@ object StreamingQueryListener extends Serializable {
    *   User-specified name of the query, null if not specified.
    * @param timestamp
    *   The timestamp to start a query.
+   * @param jobTags
+   *   The job tags that have been assigned to all the jobs started by this thread
    * @since 2.1.0
    */
   @Evolving
   class QueryStartedEvent private[sql] (
       val id: UUID,
       val runId: UUID,
       val name: String,
-      val timestamp: String)
+      val timestamp: String,
+      val jobTags: Set[String] = Set())
```

Comment:
This isn't actually binary compatible.  we should define another constructor.

Reply:
Are you referring to we leave this constructor unchanged, and create a new constructor that has this new parameter? Do we have any examples on how to do it in a binary compatible way?

Reply:
Do we have MiMa check for this class? If this is not binary compatible, it'd be ideal to let linter to fail.

Reply:
Update the code to add another constructor to keep the original signature, let me know if that's the right approach. Also just to confirm, this would only be needed for scala code right?

Reply:
This isn't compatible in Scala too ..

Reply:
Do you mind sharing what is the way to make it compatible then? Should I move `jobTags ` out of the primary constructor, and have it as a `var` instead?

Reply:
I meant that adding a parameter with a default value isn't binary compatible .. it has to be with a new constructor. see https://github.com/databricks/scala-style-guide?tab=readme-ov-file#default-parameter-values

Reply:
This isn't compatible with Scala too. Not only Java.

Reply:
ohhhh it's `private[sql]`. Okay seems fine. The constructor is private. My bad :-).

Reply:
No worries, I removed that default value anyways since it's not being used.